Persist language preference

jessbudd opened the pull request:
<!--
## Submitting a Pull Request
We love contributions and appreciate any help you can offer!
-->

## Related Issue
<!-- Include a link to the issue (e.g. Resolves #12) -->

## Summary of Changes
<!-- Briefly summarize the changes made, lists are appreciated, ideally with checklists

1. [x] Thing I fixed
1. [x] Other thing I updated
1. [x] Docs I updated
-->

sbesh91 reviewed (commented):
I like how simple this looks.  I'd like to make sure this gets looked at extensively by @mtom55 or @JamesMooreOWA

mtom55 commented:
This is looking really good... I think it's almost what we want.  

In terms of getting this to a finished state, I think we just need to be able to click on every link on the site and ensure that it doesn't toggle back to English. For example if clicking on the blog article.

jessbudd commented:
Sounds like we're comfortable moving forward with this approach. I'll add in all the other pages required to have the whole site working. 

Could someone please confirm the copy for the translation banner? The placeholder is "Unfortunately this page has not been translated into your preferred language yet. If you'd like to help us translate it, please get in touch.".  (Note: we can't put links here, only text.)

mtom55 commented:
what about "Unfortunately this page has not been translated into your preferred language yet. If you'd like to help us translate it, please join #translation on the OWA discord"

jessbudd commented on src/_data/languages_base.json:
```diff
@@ -48,7 +49,8 @@
     "siteFooterPoweredBy": "Este sitio funciona con Netlify",
-    "skipLinkText": "Saltar al contenido"
+    "skipLinkText": "Saltar al contenido",
+    "translationNotAvailable": "Lamentablemente, esta página aún no se ha traducido a su idioma preferido. Si desea ayudarnos a traducirlo, únase a #translation en OWA Discord."
```
@nickchomey @AlfonsoML could you please confirm if this translation to spanish needs any edits?

jessbudd replied:
@AlfonsoML did you have any thoughts to add here?

AlfonsoML replied:
Sorry for the delay.
I would use a mix of both, using "tu", proper genre and "Discord de OWA"

Lamentablemente, esta página aún no se ha traducido a tu idioma preferido. Si deseas ayudarnos a traducirla, únete a #translation en el Discord de OWA .

jessbudd replied:
Thanks, I'll update!

jessbudd commented on src/_data/languages_base.json:
```diff
@@ -71,6 +73,7 @@
     "siteFooterPoweredBy": "このサイトはNetlifyを利用しています",
-    "skipLinkText": "コンテンツにスキップする"
+    "skipLinkText": "コンテンツにスキップする",
+    "translationNotAvailable": "残念ながら、このページはまだご希望の言語に翻訳されていません。翻訳にご協力いただける場合は、OWA discord の #translation にご参加ください。"
```
@shuuji3 @AXT-AyaKoto could you please confirm if this translation to Japanese needs any changes?

shuuji3 replied:
@jessbudd Could you update to the following text?

```suggestion
    "translationNotAvailable": "残念ながら、このページはまだ日本語に翻訳されていません。翻訳にご協力いただける場合は、OWA Discord の #translation にご参加ください。"
```

To make it a more natural translation, I replaced "preferred language" with "Japanese" instead of a literal translation. (Also, uppercased "d" in "discord")

jessbudd replied:
Thanks so much!

jessbudd commented:
Okay, so I've:
 - added a script to auto generate all the missing pages/posts
 - added all the missing pages for `es` and `ja`
- updated all the places I could find which werent using the locale_filter
- added an auto detection for language on the homepage only, with an `/en/` version linked in the language toggle
- updated the banner copy

This is an epic number of file changes that I'd hate for anyone to have to code review all at once!

Have a click around the [deployed preview](https://deploy-preview-193--owa-production.netlify.app/) with your browser language set to Japanese, starting at the homepage. When we're happy with how everything is working, I'll break all these changes down into smaller individual PRs for reviewing/merging.

jessbudd commented:
Closed. Work has been broken into smaller separate PRs:
- https://github.com/OpenWebAdvocacy/website/pull/198
- https://github.com/OpenWebAdvocacy/website/pull/200
- https://github.com/OpenWebAdvocacy/website/pull/202
